chat-plugins/mafia.js

```diff
@@ -559,30 +679,57 @@ class MafiaTracker extends Rooms.RoomGame {
 		this.updateRoleString();
 		this.updatePlayers();
 		player.updateHtmlRoom();
-		return true;
 	}
 
-	revive(user, deadPlayer, force) {
-		if (deadPlayer in this.players) return user.sendTo(this.room, `|error|The user ${deadPlayer} is already a living player.`);
-		if (deadPlayer in this.dead) {
-			deadPlayer = this.dead[deadPlayer];
-			if (deadPlayer.treestump) delete deadPlayer.treestump;
-			if (deadPlayer.restless) delete deadPlayer.restless;
+	/**
+	 * @param {User} user
+	 * @param {string} toRevive
+	 * @param {boolean} force
+	 * @return {void}
+	 */
+	revive(user, toRevive, force = false) {
+		if (toRevive in this.players) {
+			user.sendTo(this.room, `|error|The user ${toRevive} is already a living player.`);
+			return;
+		}
+		if (toRevive in this.dead) {
+			const deadPlayer = this.dead[toRevive];
+			if (deadPlayer.treestump) deadPlayer.treestump = false;
+			if (deadPlayer.restless) deadPlayer.restless = false;
 			this.sendRoom(`${deadPlayer.safeName} was revived!`, {declare: true});
 			this.players[deadPlayer.userid] = deadPlayer;
-			this.roles.push(deadPlayer.role);
+			const targetRole = deadPlayer.role;
+			if (targetRole) {
+				this.roles.push(targetRole);
+			} else {
+				// Should never happen
+				deadPlayer.role = {
+					name: `Unknown`,
+					safeName: `Unknown`,
+					id: `unknown`,
+					alignment: 'solo',
+					image: '',
```

Comment:
should be ``null`` i think, same with the case a few lines down

(or change it in forcesetroles to be ``''``)

Reply:
I did make it `''` in all locations. Typescript would have caught it if it wasn't (and did before I fixed it).

Reply:
or well now it is anyways